src/Particle/HDFWalkerOutput.cpp

```diff
@@ -94,7 +94,8 @@ HDFWalkerOutput::~HDFWalkerOutput()
  */
 bool HDFWalkerOutput::dump(const WalkerConfigurations& W, int nblock)
 {
-  std::string FileName = myComm->getName() + hdf::config_ext;
+  std::filesystem::path FileName = myComm->getName();
+  FileName.replace_extension(hdf::config_ext);
```

Comment:
If i understand this code correctly, it may cause unexpected behavior.  The name from getName() is going to be something like "projectname.s000", and the filesystem path will assume the ".s000" part is a file extension and replace it, rather than append to it as in the original code.